### ISSUE TITLE
[java] JUnitTestsShouldIncludeAssert - false negative with `@ParameterizedTest`

### ISSUE THREAD

**Issue by ctoomey** [pmd/pmd#4114]
**Affects PMD Version:** 6.48.0

**Rule:** [JUnitTestsShouldIncludeAssert](https://pmd.sourceforge.io/pmd-6.49.0/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert)

**Description:** Rule doesn't check test methods annotated with `@ParameterizedTest` so fails to report violations therein.

**Code Sample demonstrating the issue:**

```java
import java.util.Locale;
import org.assertj.core.api.SoftAssertions;
import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

@ExtendWith({SoftAssertionsExtension.class})
class ShouldIncludeAssertTest {
  @ParameterizedTest
  @ValueSource(strings = { "racecar", "radar", "Foobar" })
  void falseNegative(String candidate) {
//    assertEquals("String isn't lowercase", candidate, candidate.toLowerCase(Locale.ENGLISH));
  }
}

```

**Expected outcome:** Flags `falseNegative`, but PMD doesn't report a violation.

**Running PMD through:** Gradle

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
